### PR TITLE
[Core][Distributed] improve logging for init dist

### PR DESCRIPTION
**Pull request by youkaichao**
Often times we see issue reports of hang during initialization. This might be caused by improper setup of ray or distributed environment.

Adding this logging will produce debugging information before the process gets stuck, helping the future debugging.

### CODE REVIEW

File: vllm/distributed/parallel_state.py [hunk 1]
```diff
@@ -45,6 +49,8 @@ def init_distributed_environment(
     local_rank: int = -1,
     backend: str = "nccl",
 ):
+    logger.info(f"{world_size=} {rank=} {local_rank=}"
+                f" {distributed_init_method=} {backend=}")
```
Comment (WoosukKwon) [inline vllm/distributed/parallel_state.py:53]: nit:
```suggestion
    logger.info(f"{world_size=} {rank=} {local_rank=} "
                f"{distributed_init_method=} {backend=}")
```

**Review by rkooo567 — APPROVED** [review]

**Comment by simon-mo** [comment]
I feel strongly that this information should go into `logger.debug` and govern by environment variable like `VLLM_LOG_LEVEL`. I previously did some pass to clean up log messages to reduce spam. Our log message should be clean and informing users. This message is used for debugging purpose only so we should set it debug level.

File: vllm/distributed/parallel_state.py [hunk 2]
```diff
@@ -45,6 +49,8 @@ def init_distributed_environment(
     local_rank: int = -1,
     backend: str = "nccl",
 ):
+    logger.info(f"{world_size=} {rank=} {local_rank=} "
```
Comment (simon-mo) [inline vllm/distributed/parallel_state.py:52]:
```suggestion
    logger.debug(f"{world_size=} {rank=} {local_rank=} "
```

**Review by simon-mo — CHANGES_REQUESTED** [review]

File: vllm/distributed/parallel_state.py (hunk 2 above)
Comment (youkaichao) [reply vllm/distributed/parallel_state.py:52]: i see the default value is debug: 

https://github.com/vllm-project/vllm/blob/5c2e66e4871917c5d59cc4a8b89ef53e690e9bd9/vllm/logger.py#L58

**Comment by youkaichao** [comment]
Given the sheer amount of issue report we receive everyday, personally I think **we should make logging verbose during vllm initialization** (and actually it is quite verbose now). When people paste the logs in issue report, that greatly help us respond/fix issue more quickly.

For logging during runtime, e.g. when vllm is processing requests, verbose logging can hurt performance. There we should be very careful, and it is better to have less logging.

**Comment by simon-mo** [comment]
PyTorch never logs things. I think our message is too verbose already. This generally gives impression of a non polished software. Imagine someone running LLM instance in a notebook and getting spammed with hundreds of messages. Similar to server we should only log very few <10 lines for initialization.

We should not log Ray worker information by default, we can set log_to_driver to false by default when initializing Ray. 

For the nccl version, we should only log a warning when it is not the common case (not 1.18 and not loaded from .cache/vllm)

For user bug report, we should ask users (through template) to turn on debug log and give us detail information, similar to the collect env ask.

**Review by simon-mo — APPROVED** [review]

**Pull request merged**